presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py

```diff
@@ -93,13 +93,40 @@ def _remove_conflicts_and_get_text_manipulation_data(
         2. Have the same indices as other results but with larger score.
         :return: List
         """
-        unique_text_metadata_elements = []
+        tmp_analyzer_results = []
         # This list contains all elements which we need to check a single result
         # against. If a result is dropped, it can also be dropped from this list
         # since it is intersecting with another result and we selected the other one.
         other_elements = analyzer_results.copy()
         for result in analyzer_results:
             other_elements.remove(result)
+
+            is_merge_same_entity_type = False
+            for other_element in other_elements:
+                if other_element.entity_type != result.entity_type:
+                    continue
+                if result.intersects(other_element) == 0:
+                    continue
+
+                other_element.start = min(result.start, other_element.start)
+                other_element.end = max(result.end, other_element.end)
+                other_element.score = max(result.score, other_element.score)
+                is_merge_same_entity_type = True
```

Comment:
What if there are two or more intersections? would it take into consideration only the first intersecting couple?
What if the intersection is as so:
-----+++++-----
--++++---------
It will basically form a new PII which is the concatenation of the two?
--+++++++-----

Can you add tests for those cases?